zebra-chain/src/parameters/network/subsidy.rs

```diff
@@ -391,8 +408,38 @@ impl ParameterSubsidy for Network {
             Network::Mainnet => NetworkUpgrade::Canopy
                 .activation_height(self)
                 .expect("canopy activation height should be available"),
-            // TODO: Check what zcashd does here, consider adding a field to `testnet::Parameters` to make this configurable.
-            Network::Testnet(_params) => FIRST_HALVING_TESTNET,
+            Network::Testnet(params) => {
+                if params.is_regtest() {
+                    FIRST_HALVING_REGTEST
+                } else if params.is_default_testnet() {
+                    FIRST_HALVING_TESTNET
+                } else {
+                    Height(
+                        self.post_blossom_halving_interval()
+                            .try_into()
+                            .expect("post blossom halving interval should fit in a Height"),
+                    )
+                }
+            }
+        }
+    }
+
+    fn post_blossom_halving_interval(&self) -> HeightDiff {
+        self.pre_blossom_halving_interval() * (BLOSSOM_POW_TARGET_SPACING_RATIO as HeightDiff)
```

Comment:
Optional/Nitpick/Subjective: Could we calculate this when building `testnet::Parameters` and add it as a field?